Source/FieldSolver/FiniteDifferenceSolver/EvolveB.cpp

```diff
@@ -197,6 +211,223 @@ void FiniteDifferenceSolver::EvolveBCartesian (
+            auto borrowing_dim_area = borrowing_dim.area.data();
+
+            auto const &borrowing_inds = (*borrowing[idim])[mfi].inds.data();
+            auto const &borrowing_size = (*borrowing[idim])[mfi].size.array();
+            auto const &borrowing_inds_pointer = (*borrowing[idim])[mfi].inds_pointer.array();
+
+            // Extract tileboxes for which to loop
+            Box const &tb = mfi.tilebox(Bfield[idim]->ixType().toIntVect());
+
+            //Take care of the unstable cells
+            amrex::ParallelFor(tb,
+
+            [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+
+                if (S(i, j, k) <= 0) return;
+
+                if (!(flag_info_cell_dim(i, j, k) == 0))
+                    return;
+
+                Venl_dim(i, j, k) = Rho(i, j, k) * S(i, j, k);
+                amrex::Real rho_enl;
+
+                // First we compute the rho of the enlarged face
+                for (int offset = 0; offset<borrowing_size(i, j, k); offset++) {
+                    int ind = borrowing_inds[*borrowing_inds_pointer(i, j, k) + offset];
+                    auto vec = FaceInfoBox::uint8_to_inds(borrowing_dim_neigh_faces[ind]);
+                    int ip, jp, kp;
+                    if(idim == 0){
+                        ip = i;
+                        jp = j + vec(0);
+                        kp = k + vec(1);
+                    }else if(idim == 1){
+                        ip = i + vec(0);
+                        jp = j;
+                        kp = k + vec(1);
+                    }else{
+                        ip = i + vec(0);
+                        jp = j + vec(1);
+                        kp = k;
+                    }
+
+                    Venl_dim(i, j, k) += Rho(ip, jp, kp) * borrowing_dim_area[ind];
+
+                }
+
+                rho_enl = Venl_dim(i, j, k) / S_mod(i, j, k);
+
+                for (int offset = 0; offset < borrowing_size(i, j, k); offset++) {
+                    int ind = borrowing_inds[*borrowing_inds_pointer(i, j, k) + offset];
+                    auto vec = FaceInfoBox::uint8_to_inds(borrowing_dim_neigh_faces[ind]);
+                    int ip, jp, kp;
+                    if(idim == 0){
+                        ip = i;
+                        jp = j + vec(0);
+                        kp = k + vec(1);
+                    }else if(idim == 1){
+                        ip = i + vec(0);
+                        jp = j;
+                        kp = k + vec(1);
+                    }else{
+                        ip = i + vec(0);
+                        jp = j + vec(1);
+                        kp = k;
+                    }
+
+                    Venl_dim(ip, jp, kp) += rho_enl * borrowing_dim_area[ind];
+
+                }
+
+                B(i, j, k) = B(i, j, k) - dt * rho_enl;
+
+            });
+
+            //Take care of the stable cells
+            amrex::ParallelFor(tb,
+
+            [=] AMREX_GPU_DEVICE(int i, int j, int k) {
```

Comment:
I think it could be easier to read the `ParallelFor` in one single line (since only one single `AMREX_GPU_DEVICE` follows):
```suggestion
            amrex::ParallelFor(tb, [=] AMREX_GPU_DEVICE(int i, int j, int k) {
```